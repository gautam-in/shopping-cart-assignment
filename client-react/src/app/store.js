import { configureStore } from '@reduxjs/toolkit';

import bannerReducer from '../features/banner/BannerSlice'
import categoryReducer from '../features/categories/CategorySlice'
import productReducer from '../features/productList/ProductListSlice'
import cartReducer from '../features/cart/CartSlice'

export const store = configureStore({
  reducer: {
    banner: bannerReducer,
    category: categoryReducer,
    product: productReducer,
    cart: cartReducer
  },
});

