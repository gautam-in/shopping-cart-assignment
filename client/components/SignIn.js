import React from 'react'
import Form from './styles/Form';
import useForm from '../utils/useForm';
import useLocalStorage from '../utils/useLocalStorage';
import { useRouter }  from 'next/router'
import Link from 'next/link';



export default function SignIn() {
  const { inputs, handleChange, resetForm } = useForm({
    email: '',
    password: '',
  });
  const [userInfo, setUserInfo] = useLocalStorage("userinfo", "");
  const router = useRouter()
  async function handleSubmit(e) {
    e.preventDefault(); // stop the form from submitting
    setUserInfo(inputs)
    alert('Successfully Loggeed In !')
    router.push('/')
    router.reload()
    resetForm();
    // Send the email and password to the graphqlAPI
  }

  if(userInfo && userInfo.email) {
      return <>
      <h4>Welcome! {userInfo.email}</h4>
       <p>Visit <button><Link href="/products">Products üç®</Link></button></p>
      </>
  }

  return (
    <Form method="POST" onSubmit={handleSubmit}>
      <h2>Sign Into Your Account</h2>
      <fieldset>
        <label htmlFor="email">
          Email
          <input
            type="email"
            name="email"
            placeholder="Your Email Address"
            autoComplete="email"
            value={inputs.email}
            onChange={handleChange}
          />
        </label>
        <label htmlFor="password">
          Password
          <input
            type="password"
            name="password"
            placeholder="Password"
            autoComplete="password"
            value={inputs.password}
            onChange={handleChange}
          />
        </label>
        <button type="submit">Sign In!</button>
      </fieldset>
    </Form>
  );
}