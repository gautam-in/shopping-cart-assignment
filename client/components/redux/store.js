import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";

import addItemReducer from "./addItem/addItemReducer";
import rootReducer from "./rootReducer";

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);
/* const store = createStore(addItemReducer, composeWithDevTools()); */

export default store;
