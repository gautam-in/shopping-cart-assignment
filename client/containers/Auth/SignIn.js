import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import "./Auth.scss";
import AuthContext from "../../AuthContext";

const SignIn = () => {
  const { toggleUserAuthentication } = useContext(AuthContext);
  const history = useHistory();
  const intialState = {
    email: "",
    password: "",
  };
  const [userDetails, setUserDetails] = useState(intialState);
  const [validation, setValidation] = useState("");

  const onFormSubmit = (e) => {
    e.preventDefault();
    const email = sessionStorage.getItem("email");
    const pwd = sessionStorage.getItem("password");
    if (userDetails.email === email && userDetails.password === pwd) {
      toggleUserAuthentication();
      sessionStorage.setItem("status", "logged-in");
      history.push("/");
    } else {
      setValidation("Invalid credentials");
    }
  };

  return (
    <div className="sign-in">
      <aside>
        <h1>SignIn</h1>
        <p>Get access to your orders, wishlist and recommedations</p>
      </aside>
      <section className="register-form">
        <form onSubmit={onFormSubmit}>
          <input
            placeholder="Email"
            value={userDetails.email}
            required
            onChange={(event) =>
              setUserDetails({ ...userDetails, email: event.target.value })
            }
          />
          <input
            placeholder="Password"
            type="password"
            value={userDetails.password}
            required
            onChange={(event) =>
              setUserDetails({ ...userDetails, password: event.target.value })
            }
          />
          <p
            style={{
              color: "red",
              textAlign: "center",
            }}
          >
            {validation}
          </p>
          <button>SignIn</button>
        </form>
      </section>
    </div>
  );
};

export default SignIn;
