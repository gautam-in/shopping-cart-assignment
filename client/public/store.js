import React, {
  useReducer,
  createContext,
  useContext,
  useCallback,
} from "react";

const StateContext = createContext();
const DispatchContext = createContext();

export const StoreProvider = ({ children }) => {
  const reducer = useCallback((state, action) => {
    const newState = { ...state };
    const productIndex = state.cart?.items?.findIndex(
      (val) => val.id == action.payload.id
    );
    console.log("state", state, action.payload.id, productIndex);
    switch (action.type) {
      case "ADD_TO_CART":
        if (productIndex < 0) {
          add(newState, action.payload);
        } else {
          increament(newState, productIndex);
        }
        return newState;
      case "INCREAMENT":
        increament(newState, productIndex);
        return newState;
      case "DECREAMENT":
        decreament(newState, productIndex);
        return newState;
      case "REMOVE":
        remove(newState, productIndex);
        return newState;
      case "USER":
        return { ...state, ...action.payload };
      case "TOGGLE_CART":
        return { ...state, ...action.payload };
      case "CLOSE_CART":
        return { ...state, ...action.payload };
      case "DO_CART_EMPTY":
        return { ...state, ...action.payload };
      default:
        throw new Error("unknown Wrror");
    }
  }, []);
  // const UserCache = localStorageLoggedInuser();
  const [cartState, dispatch] = useReducer(reducer, {
    cart: { items: [] },
    user: {},
    cartOpen: false,
  });

  return (
    <DispatchContext.Provider value={dispatch}>
      <StateContext.Provider value={cartState}>
        {children}
      </StateContext.Provider>
    </DispatchContext.Provider>
  );
};

function increament(newState, productIndex) {
  newState.cart.items[productIndex].quantity += 1;
}
function decreament(newState, productIndex) {
  newState.cart.items[productIndex]?.quantity > 1
    ? (newState.cart.items[productIndex].quantity -= 1)
    : remove(newState, productIndex);
}
function remove(newState, productIndex) {
  const filteredItems = newState.cart.items.filter(
    (item) => item.id != newState.cart.items[productIndex].id
  );
  newState.cart.items = [...filteredItems];
}
function add(newState, payload) {
  newState.cart.items.push(payload);
}

export const useStore = () => {
  const cartState = useContext(StateContext);
  const dispatch = useDispatchStore();
  const { cart, user, cartOpen } = cartState;
  const { items } = cart;
  const total = items.reduce((acc, item) => {
    return acc + item.quantity * item.price;
  }, 0);
  const totalquantity = items.reduce((acc, item) => {
    return acc + item.quantity;
  }, 0);
  const setToggleCart = () => {
    dispatch({
      type: "TOGGLE_CART",
      payload: { cartOpen: !cartOpen },
    });
  };
  const setCloseCart = () => {
    dispatch({
      type: "CLOSE_CART",
      payload: { cartOpen: false },
    });
  };
  const doCartEmpty = () => {
    dispatch({
      type: "DO_CART_EMPTY",
      payload: { cart: { items: [] } },
    });
  };
  return {
    items,
    total,
    totalquantity,
    user,
    cartOpen,
    setToggleCart,
    setCloseCart,
    doCartEmpty,
  };
};
export const useDispatchStore = () => useContext(DispatchContext);
