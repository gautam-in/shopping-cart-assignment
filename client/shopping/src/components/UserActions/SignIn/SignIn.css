.loginDes{
    height:3em;
    font-size:1.2em;
}
.loginDes>*{
    height: 2em;
    margin: 10% 0% 0% 10%;
}
.loginDes>header{
    font-weight:bolder;
    font-size:1.5em;
    margin-top: 0;
}

  input[type="text"],input[type="password"] {
    box-sizing: border-box;
    width: 100%;
    height: calc(4em + 1px);
    margin: 0 0 1em;
    padding: 0.5em;
    border:0px;
    border-bottom: 1px solid #ccc;
    background: #fff;
    resize: none;
    outline: none;
  }
  
  input[type="text"][required]:focus,
  input[type="password"][required]:focus {
    border-color: #00bafa;
  }
  input[type="text"][required]:focus + label[placeholder]:before, 
  input[type="password"][required]:focus + label[placeholder]:before {
    color: #00bafa;
  }
  input[type="text"][required]:focus + label[placeholder]:before,
  input[type="text"][required]:valid + label[placeholder]:before,
  input[type="password"][required]:focus + label[placeholder]:before,
  input[type="password"][required]:valid + label[placeholder]:before {
    transition-duration: .2s;
    transform: translate(0, -1.5em) scale(0.9, 0.9);
  }
  
  input[type="text"][required]:invalid + label[placeholder][alt]:before,
  input[type="password"][required]:invalid + label[placeholder][alt]:before {
    content: attr(alt);
  }
  
  input[type="text"][required] + label[placeholder],
  input[type="password"][required] + label[placeholder] {
    display: block;
    pointer-events: none;
    line-height: 1em;
    margin-top: calc(-3em - 2px);
    margin-bottom: calc((3em - 1em) + 2px);
  }
  
  input[type="text"][required] + label[placeholder]:before,
  input[type="password"][required] + label[placeholder]:before {
    content: attr(placeholder);
    display: inline-block;
    padding: 0 2px;
    color: #898989;
    white-space: nowrap;
    transition: 0.3s ease-in-out;
    background-image: linear-gradient(to bottom, #fff, #fff);
    background-size: 100% 5px;
    background-repeat: no-repeat;
    background-position: center;
  }
  .loginform{
      margin-top: 5%;
  }
  .loginform>*{
      width: 90% !important;
  }
  .loginform>button{
    padding: 1em;
  }