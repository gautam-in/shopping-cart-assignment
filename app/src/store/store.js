import { compose, createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger'
import { rootReducer } from "./root-reducer";

const middleware = [logger];

const composedEnhancers = compose(applyMiddleware(...middleware))

export const store = createStore(rootReducer,undefined,composedEnhancers)