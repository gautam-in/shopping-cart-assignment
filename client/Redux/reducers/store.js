import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";
import { cartReducer } from "./cartReducer";
import { categoryReducer } from "./categoryReducer";
import { errorReducer } from "./errorReducer";
import { userReducer } from "./userReducer";

const store = createStore(
  combineReducers({
    cart: cartReducer,
    categories: categoryReducer,
    user: userReducer,
    error: errorReducer
  }),
  applyMiddleware(thunk)
);

export default store;
