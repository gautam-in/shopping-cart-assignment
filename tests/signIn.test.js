import store from "../Redux/reducers/store";
import SignIn from "../pages/signIn";


import {mount} from "enzyme";
import { Provider } from "react-redux";


describe("SignInPage", () => {
  const wrapper = mount(
    <Provider store={store}>
      <SignIn />
    </Provider>
  );
  test("SignInPage should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
