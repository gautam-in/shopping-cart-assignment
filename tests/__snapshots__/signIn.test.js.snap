// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInPage SignInPage should match snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <signIn>
    <SignInComponent>
      <FormStyle__FormContainer>
        <section
          className="FormStyle__FormContainer-sc-wpd7la-0 iccOel"
        >
          <FormStyle__FormLeft>
            <div
              className="FormStyle__FormLeft-sc-wpd7la-1 jLIMxq"
            >
              <h2>
                Login
              </h2>
              <p>
                Get access to your Orders, Wishlist and Recommendations
              </p>
            </div>
          </FormStyle__FormLeft>
          <FormStyle__FormRight>
            <div
              className="FormStyle__FormRight-sc-wpd7la-2 grYBdk"
            >
              <form
                onSubmit={[Function]}
              >
                <InputStyle>
                  <div
                    className="InputStyle-sc-gzj6cq-0 hqkFkK"
                  >
                    <input
                      id="email"
                      name="email"
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value=""
                    />
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                  </div>
                </InputStyle>
                <InputStyle>
                  <div
                    className="InputStyle-sc-gzj6cq-0 hqkFkK"
                  >
                    <input
                      id="password"
                      name="password"
                      onChange={[Function]}
                      required={true}
                      type="password"
                      value=""
                    />
                    <label
                      htmlFor="Password"
                    >
                      Password
                    </label>
                  </div>
                </InputStyle>
                <InputStyle__LargeButtonStyle>
                  <div
                    className="InputStyle__LargeButtonStyle-sc-gzj6cq-1 ifaKse"
                  >
                    <input
                      type="submit"
                      value="Login"
                    />
                  </div>
                </InputStyle__LargeButtonStyle>
              </form>
            </div>
          </FormStyle__FormRight>
        </section>
      </FormStyle__FormContainer>
    </SignInComponent>
  </signIn>
</Provider>
`;
